File: apps/backend/app/schemas/company_profile.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class CompanyProfileRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    description: str | None = None
    industry: str | None = None
    product_modules: list[str] = Field(default_factory=list)
    target_users: list[str] = Field(default_factory=list)
    business_goals: list[str] = Field(default_factory=list)
    technical_stack: list[str] = Field(default_factory=list)
    competitors: list[str] = Field(default_factory=list)
    risk_tolerance: str
    engineering_capacity: str
    raw_profile: dict[str, Any] | None = None
    created_at: datetime
    updated_at: datetime

File: apps/frontend/src/types/company-profile.ts
export interface CompanyProfile {
  id: string;
  name: string;
  description?: string | null;
  industry?: string | null;
  product_modules: string[];
  target_users: string[];
  business_goals: string[];
  technical_stack: string[];
  competitors: string[];
  risk_tolerance: string;
  engineering_capacity: string;
  raw_profile?: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
}
